Feature/fv/fsm encap

favillat opened the pull request:
Added getter and setter for fsm 
-replaced reading and writting with their corresponding functions

favillat committed:
Added fc_state_update and get_fc_state

favillat committed:
ALL state transitions now fc_state_update
-replaced transisiotns
-replaced readings
-yay encapsulation

ETSells commented on app/flight/appa/rev2/flash_appa.c:
```diff
@@ -302,7 +301,7 @@ uint8_t idx = 0; /* current index in the buffer */
 memset(buffer, 0, sensor_frame_size);
 buffer[0] = 1; /* save bit */
-buffer[1] = flight_computer_state;
+buffer[1] = get_fc_state();
```
yeah like this!

ETSells commented on app/flight/appa/rev2/fsm_appa.c:
```diff
@@ -76,7 +93,7 @@ if( *flash_status == FLASH_PRESET_NOT_FOUND )
 	}
 
-flight_computer_state = FC_STATE_IDLE;
+fc_state_update( FC_STATE_IDLE );
```
yes! we can't enforce the encapsulation in this file, but we can at least make sure it's standard to not edit it manually.

ETSells commented on app/flight/appa/rev2/main.c:
```diff
@@ -269,6 +267,17 @@ appa_fsm
 	&sensor_status
 	);
 
+bool fc_state_update
```
remove these two function prototypes. they're in another function and likely would break the build if another issue wasnt breaking it first.

ETSells reviewed (changes requested):
A few critical build errors need to be resolved.

ETSells commented:
@favillat On your next commit, make sure you watch the output of that `FCF Stability Check`. If it goes green, you're good. If it's red, you have a build error you should investigate.

favillat committed:
pr changes
-removed fsm function prototypes from main
-undid dual-deploy changes
-changed fsm update state funtion signature to VOID
-updated flash memcopy so it doesnt break everything

favillat committed:
Updated get_state signature in main.h
-forgor to switch from bool to void

favillat commented:
I built locally and got no errors, this is going to sound so silly but I totally forgot that was something I should be doing...I was a little confused at first as to why I failed the stability check

ETSells commented on app/flight/appa/rev2/fsm_appa.c:
```diff
@@ -41,7 +41,22 @@ extern volatile uint32_t debug_delta;
+        }
+    }
 
+FLIGHT_COMP_STATE_TYPE get_fc_state()
```
coding standards: include procedure description

ETSells reviewed (approved):
LGTM! Please just make the requested standards fixes. The procedure descriptions in particular are important to me

favillat committed:
added function descriptions, procedure fixes.
-fc_state_update
-get_fc_state functions
